### PR TITLE
Feature/basic stats for management panel

### PR DESCRIPTION
**Pull request by Jakedavies**
I designed this around #68 being ready, since it hard to test otherwise, do not merge until that is merged into master and we rebase this on top

### CODE REVIEW

File: cg/test/management/stats.api.spec.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+jest.mock('../../src/utils/blockchain/web3-provider-factory');
+
+const winston = require('winston');
+
+const { initResources, fetch, closeResources } = require('../utils');
+const { deployContract } = require('../blockchain-setup');
+const { db } = require('../../src/db');
+const { managementJWT, subjectJWT } = require('../../src/utils/jwt');
+const { setProcessors } = require('../../src/utils/blockchain');
```
Comment (sindelio) [inline cg/test/management/stats.api.spec.js]: Just a reminder that setProcessors is meant to facilitate tests and should not be used in the actual API

File: cg/test/management/stats.api.spec.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        data: {
+          controller: {
+            consented: 1,
+            unconsented: 1,
+            total: 2
+          },
+          processors: {
+            '201': {
+              // I was expecting this to be 1 originally, but it appears the be 2 because requesting erasure does not cascade, should it?
+              consented: 2
```
Comment (sindelio) [inline cg/test/management/stats.api.spec.js]: A processor can't have consent for a subject's data that was erased, that's misleading. I think we should address this in the erase-data endpoint

Comment (sindelio) [inline cg/test/management/stats.api.spec.js]: Also, I can't think of other tests for this endpoint

**Comment by Jakedavies** [comment]
@sindelio I will create an issue to correct the behavior of the erase-data functionality since it is probably out of scope for this PR

File: cg/src/domains/management/stats/stats.service.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+class StatsService {
+  constructor(database = db) {
+    this.db = database;
+  }
+
+  async stats() {
+    const [result] = await db('subjects')
+      .select(db.raw('count(subjects.id) as consented'))
+      .join('subject_keys', 'subjects.id', 'subject_keys.subject_id');
```
Comment (maksymilian-majer) [inline cg/src/domains/management/stats/stats.service.js]: We could have one query instead of two to left join on keys and count when not null (use `COALESCE` with `CASE` or something like it), but I don't expect a huge performance gain.

File: cg/test/management/stats.api.spec.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  addressesToInsert.forEach(a => expect(addresses).toContainEqual(a));
+});
+
+beforeEach(() => {
+  process.env.MODE = VALID_RUN_MODES.CONTROLLER;
+});
+afterAll(closeResources);
+
+describe('Stats endpoint', () => {
+  it('Should display the stats for the current system state', async () => {
```
Comment (maksymilian-majer) [inline cg/test/management/stats.api.spec.js]: I suggest a test case (at least one) with no data first to see if we get zero's. I suspect a 500 error when there are no subjects. Maybe also some edge cases when consented is 0 and total is > 0, etc.

File: cg/src/domains/management/stats/stats.service.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    const consentedSubjectCount = parseInt(result.consented, 10);
+
+    const [result2] = await db('subjects').select(db.raw('count(subjects.id) as total'));
+    const totalSubjectCount = parseInt(result2.total, 10);
```
Comment (maksymilian-majer) [inline cg/src/domains/management/stats/stats.service.js]: If there are no subjects I suspect there will be an error thrown on `result2.total`.

Comment (maksymilian-majer) [reply cg/src/domains/management/stats/stats.service.js]: Never mind – count will always return result even if zero.
Now there's a test for this :)
